src/features/instrument/Instrument.tsx

```diff
@@ -0,0 +1,155 @@
+import { BASS_CHORDS, GAMMA_NOTES, MARIO_SAMPLES, MELODY_CHORDS } from '../../constants/notes'
+import { useRef } from 'react';
+const DogSample = require("../../assets/dog-sample.mp3");
+
+const keyStyles = {
+  paddingTop: '4em',
+  paddingBottom: '4em',
+  width: '5em'
+};
+const mediumVolume = new Volume(-15);
+const sampler = new Sampler({
+  E4: DogSample,
+}).chain(mediumVolume, Destination);
+const polySynthSquare = new PolySynth(Synth, {
+  oscillator: {
+    type: "square",
+  },
+}).chain(new Volume(-25), Destination);
+
+export const Instrument = () => {
+  // @TODO: separate to piaoa nd predefined isntruments
+
+  const isTransportStarted = useRef(false);
+  const melodyPart = useRef<Part | null>(null);
+  const bassPart = useRef<Part | null>(null);
+  const noise = useRef<Noise | null>(null);
+
+  function playNote(note: string) {
+    polySynthSquare.triggerAttackRelease(note, "8n");
+    sampler.triggerAttack(note);
+  }
+
+  function playPart(part: Part) {
+    if (!isTransportStarted.current) {
+      Transport.toggle();
+      isTransportStarted.current = true;
+      Transport.bpm.value = 132;
+    }
+
+    part.start(0);
+    part.loop = true;
+    part.loopEnd = "4m";
+  }
+
+  // @TODO: maybe move to separate components, maybe same file
+  function toggleMelody() {
+    if (melodyPart.current) {
+      melodyPart.current.mute = !melodyPart.current.mute;
+    } else {
+      const polySynthSaw = new PolySynth(Synth, {
+        oscillator: {
+          type: "fatsawtooth",
+        },
+        envelope: {
+          attack: 0.01,
+          release: 0.4,
+        },
+      }).chain(mediumVolume, Destination);
+
+      melodyPart.current = new Part((time, chord) => {
+        sampler.triggerAttackRelease(chord.note, chord.duration, time);
+        polySynthSaw.triggerAttackRelease(
+          chord.note,
+          chord.duration,
+          time
+        );
+      }, MELODY_CHORDS);
+
+      playPart(melodyPart.current);
+    }
+  }
+
+  function toggleBass() {
+    if (bassPart.current) {
+      bassPart.current.mute = !bassPart.current.mute;
+    } else {
+      bassPart.current = new Part((time, chord) => {
+        polySynthSquare.triggerAttackRelease(
+          chord.note,
+          chord.duration,
+          time
+        );
+      }, BASS_CHORDS);
+      playPart(bassPart.current);
+    }
+  }
+
+  function toggleNoise() {
+    if (noise.current) {
+      noise.current.mute = !noise.current.mute;
+    } else {
+      noise.current = new Noise("pink").start();
+      const filter = new AutoFilter({
+        frequency: "8m",
+      }).chain(new Volume(-20), Destination);
+      noise.current.connect(filter);
+      filter.start();
+    }
+  }
+
+  // @TODO: check arent a beter way to write each thing now (always create new player?), also check updates in tone
+  function playSample(sampleName: string) {
+    new Player({
+      url: sampleName,
+      autostart: true,
+    }).chain(mediumVolume, Destination);
```

Comment:
Seems like we could initialize the samples onload and then start/stop them vs creating new ones each time according to the docs?

https://tonejs.github.io/docs/14.7.77/Player#start